statline_bq/utils.py

```diff
@@ -75,6 +75,109 @@ def check_v4(id: str, third_party: bool = False) -> str:
     return odata_version
 
 
+def get_metadata_cbs(urls: dict, odata_version: str):
+    if odata_version == "v3":
+        target_url = "?".join((urls["TableInfos"], "$format=json"))
+        meta = requests.get(target_url).json()["value"][0]
+    elif odata_version == "v4":
+        meta = requests.get(urls["Properties"]).json()
+    else:
+        raise ValueError("odata version must be either 'v3' or 'v4'")
+    return meta
+
+
+def get_latest_folder(gcs_folder: str, gcp: GcpProject) -> Union[str, None]:
+    # Initialize Google stroage Client
```

Comment:
While I do appreciate the many comments you write to make the code readable, I think it is better still to stick to the Python guidelines:
- write a docstring that explains the function
- focus on explaining the _why_, not the _what_. The latter should be obvious from the code itself.

So for example, the docstring could be:

```Get latest upload date for a given CBS dataset in gcs_folder.

Datasets from CBS are uploaded on GCS in folders `project-id/cbs/[v3|v4]/dataset-id/YYYYMMDD`. 
These folders are traversed and the most recent date is retrieved.

Args
------

gcs_folder (str): path of CBS dataset ID excluding the date part
odata_version (str): 'v3' or 'v4'

Returns
----------
json: metadata of dataset
```